ui/app/components/copy-button.js

```diff
@@ -0,0 +1,18 @@
+import Component from '@ember/component';
+import { run } from '@ember/runloop';
+import { task } from 'ember-concurrency';
+
+export default Component.extend({
+  classNames: ['copy-button'],
+
+  clipboardText: null,
+  state: null,
+
+  indicateSuccess: task(function*() {
+    this.set('state', 'success');
+
+    yield run.later(() => {
+      this.set('state', null);
+    }, 2000);
```

Comment:
Unfortunately, I think this implementation is still subject to the same bug. I just reproduced it on the Netlify preview, at least:

![image](https://user-images.githubusercontent.com/68917/61233746-466e0f00-a6ff-11e9-80f6-327afbb21791.png)

Here's one way to avoid the issue:

```js
import { task, timeout} from 'ember-concurrency';

export default Component.extend({
  // …
  indicateSuccess: task(function*() {
    this.set('state', 'success');
    yield timeout(2000);
    this.set('state', null);
  }).restartable(),
});
```

This works because on cancelation, the `timeout` is aborted, and the rest of the task is not executed.

(I also added `restartable` to make sure we don't have multiple concurrent version of this task running if the user clicks a few times.)

Reply:
goodness, yes, much better, thanks!